r/tests/testthat/test-dplyr-funcs-type.R

```diff
@@ -845,6 +854,15 @@ test_that("as.Date() converts successfully from date, timestamp, integer, char a
     warning = TRUE
   )
 
+  # strptime does not support a partial format -
+  # TODO revisit once - https://issues.apache.org/jira/browse/ARROW-15813
+  expect_error(
+    test_df %>%
+      arrow_table() %>%
+      mutate(date_char_ymd = as_date(character_ymd_var)) %>%
+      collect()
+  )
```

Comment:
Is the error here coming from C++ only (and is that why you're not asserting what the error is)? If yes to both: can we add something to the comment. We should be clear (to our future selves) why we are going against our pattern that one should always assert what the error is when using `expect_error`.

Reply:
Done